File: Leetcode/3000 - Maximum Area of Longest Diagonal Rectangle/solution.cpp
int areaOfMaxDiagonal(vector<vector<int>>& dimensions) {
    int resDiag = -1, resArea = -1;

    for (const auto& d : dimensions) {
        int len = d[0], wid = d[1];
        int currDiag = len * len + wid * wid;
        int currArea = len * wid;

        if (currDiag > resDiag || (currDiag == resDiag && currArea > resArea)) {
            resDiag = currDiag;
            resArea = currArea;
        }
    }

    return resArea;
}

File: Leetcode/3000 - Maximum Area of Longest Diagonal Rectangle/solution.js
var areaOfMaxDiagonal = function(dimensions) {
    let resDiag = -1, resArea = -1;

    for (const d of dimensions) {
        const len = d[0], wid = d[1];
        const currDiag = len * len + wid * wid;
        const currArea = len * wid;

        if (currDiag > resDiag || (currDiag === resDiag && currArea > resArea)) {
            resDiag = currDiag;
            resArea = currArea;
        }
    }

    return resArea;
};